src/providers/pronouns/Pronouns.cpp

```diff
@@ -0,0 +1,64 @@
+#include "providers/pronouns/Pronouns.hpp"
+
+#include "Application.hpp"
+#include "common/QLogging.hpp"
+#include "providers/pronouns/alejo/PronounsAlejoApi.hpp"
+#include "providers/pronouns/UserPronouns.hpp"
+
+#include <mutex>
+#include <string>
+#include <unordered_map>
+
+namespace chatterino::pronouns {
+
+void Pronouns::fetch(const QString &username,
+                     const std::function<void(UserPronouns)> &callbackSuccess,
+                     const std::function<void()> &callbackFail)
+{
+    // Only fetch pronouns if we haven't fetched before.
+    {
+        std::shared_lock<std::shared_mutex> lock(this->mutex);
```

Comment:
Nit: The template argument could be removed, as it is automatically deduced (same for the other locations).